dblookupext/esdtSupply/esdtSuppliesProcessor.go

```diff
@@ -39,11 +42,19 @@ func NewSuppliesProcessor(
 
 	logsGet := newLogsGetter(marshalizer, logsStorer)
 	logsProc := newLogsProcessor(marshalizer, suppliesStorer)
+	supplyCorrectionProc := newSupplyCorrectionProcessor(shardID, logsProc)
 
-	return &suppliesProcessor{
+	sp := &suppliesProcessor{
 		logsProc: logsProc,
 		logsGet:  logsGet,
-	}, nil
+	}
+
+	err := supplyCorrectionProc.applySupplyCorrections(supplyCorrections)
```

Comment:
This works only if a restart happens after the processing of th BlockNonce set in the  SupplyCorrections. 
If this new version is started before the BlockNonce, another restart is required, right?